cluster/src/test/java/com/hartwig/pipeline/smoke/SmokeTest.java

```diff
@@ -59,48 +58,67 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void runFullPipelineAndCheckFinalStatus() throws Exception {
-        String apiUrl = "http://api.pilot-1";
-        PipelineMain victim = new PipelineMain();
-        String version = System.getProperty("version");
-        String runId = "smoke-" + noDots(version);
-        String setName = "smoke_test";
+    public void tumorReference() throws Exception {
+        runFullPipelineAndCheckFinalStatus("tumor-reference", PipelineStatus.QC_FAILED);
+    }
+
+    @Test
+    public void tumorOnly() throws Exception {
+        runFullPipelineAndCheckFinalStatus("tumor", PipelineStatus.SUCCESS);
+    }
 
-        String privateKeyPath = workingDir() + "/google-key.json";
-        Arguments arguments = Arguments.defaultsBuilder(Arguments.DefaultsProfile.DEVELOPMENT.toString())
-                .privateKeyPath(privateKeyPath)
-                .uploadPrivateKeyPath(privateKeyPath)
+    @Test
+    public void referenceOnly() throws Exception {
+        runFullPipelineAndCheckFinalStatus("reference", PipelineStatus.SUCCESS);
+    }
+
+    public void runFullPipelineAndCheckFinalStatus(final String inputMode, final PipelineStatus expectedStatus) throws Exception {
+        PipelineMain victim = new PipelineMain();
+        String version = noDots(version(inputMode));
+        String setName = inputMode + "-" + version;
+        final String fixtureDir = "smoke_test/" + inputMode + "/";
+        final ImmutableArguments.Builder builder = Arguments.defaultsBuilder(Arguments.DefaultsProfile.DEVELOPMENT.toString())
+                .sampleJson(Resources.testResource(fixtureDir + "samples.json"))
                 .cloudSdkPath(CLOUD_SDK_PATH)
-                .setId(setName)
-                .runId(runId)
+                .setId(inputMode)
+                .runId(version)
                 .runGermlineCaller(false)
-                .sbpApiRunId(SBP_RUN_ID)
-                .sbpApiUrl(apiUrl)
-                .rclonePath(RCLONE_PATH)
-                .rcloneGcpRemote(GCP_REMOTE)
-                .rcloneS3RemoteDownload("s3")
                 .cleanup(true)
-                .outputBucket("services-pipeline-output-pilot-1")
-                .context(Context.SERVICES)
-                .build();
+                .outputBucket("smoketest-pipeline-output-pilot-1")
+                .context(Context.DIAGNOSTIC);
+
+        final String username = System.getProperty("user.name");
+        if (username.equals("root")) {
+            String privateKeyPath = workingDir() + "/google-key.json";
+            builder.privateKeyPath(privateKeyPath).uploadPrivateKeyPath(privateKeyPath);
+        } else {
+            builder.cloudSdkPath(String.format("/Users/%s/google-cloud-sdk/bin", username));
```

Comment:
The Hartwig Linux Users Group may be stirred to contribute a patch for this.

Reply:
apologies. i can only think of one possible solution https://www.apple.com/ca/macbook-pro/